pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -1121,6 +1133,82 @@ func (l *LibvirtDomainManager) getDomainSpec(dom cli.VirDomain) (*api.DomainSpec
 	return domainSpec, err
 }
 
+func removePreviousMemoryDump(dir string) {
+	files, err := ioutil.ReadDir(dir)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to remove older memory dumps")
+		return
+	}
+	for _, file := range files {
+		if strings.Contains(file.Name(), "memory.dump") {
+			err = os.Remove(filepath.Join(dir, file.Name()))
+			if err != nil {
+				log.Log.Reason(err).Errorf("failed to remove older memory dumps")
+			}
+		}
+	}
+}
+
+func (l *LibvirtDomainManager) completeMemoryDump(err error) {
+	defer func() { <-l.memoryDumpInProgress }()
+	l.lastMemoryDump.err = err
+	l.lastMemoryDump.completed = true
+}
```

Comment:
why not report the memory dump completed on the domain using the same pattern live migration uses?